common/util/file_util.cc

```diff
@@ -46,6 +48,34 @@ absl::string_view Stem(absl::string_view filename) {
              : filename.substr(0, last_dot_pos);
 }
 
+bool UpwardFileSearch(std::string start, std::string filename,
+                      std::string& result) {
+  /* Convert to absolute path */
+  char absolute_path[PATH_MAX];
+  if (realpath(std::string(start).c_str(), absolute_path) < 0) {
+    return false;
+  }
+  start = std::string(absolute_path);
+
+  std::vector<absl::string_view> search = absl::StrSplit(start, '/');
+
+  /* Push back linter configuration file name */
+  search.push_back(filename);
+
+  while (search.size() > 1) {
+    std::string candidate = absl::StrJoin(search,  "/");
+
+    if (access(candidate.c_str(), R_OK) != -1) {
```

Comment:
It probably makes sense to consider using `std:: filesystem`, but that would change the std requirements and as it wasn't used in other helpers I didn't use it here.